examples/C++/DDS/BasicConfigurationExample/BasicConfiguration_main.cpp

```diff
@@ -0,0 +1,212 @@
+            option::printUsage(fwrite, stdout, usage, columns);
+            return 1;
+        }
+
+        argc -= (argc > 0);
+        argv += (argc > 0); // skip program name argv[0] if present
+        --argc; ++argv; // skip pub/sub argument
+        option::Stats stats(usage, argc, argv);
+        std::vector<option::Option> options(stats.options_max);
+        std::vector<option::Option> buffer(stats.buffer_max);
+        option::Parser parse(usage, argc, argv, &options[0], &buffer[0]);
+
+        if (parse.error())
+        {
+            option::printUsage(fwrite, stdout, usage, columns);
+            return 1;
+        }
+
+        if (options[optionIndex::HELP])
+        {
+            option::printUsage(fwrite, stdout, usage, columns);
+            return 0;
+        }
+
+        for (int i = 0; i < parse.optionsCount(); ++i)
+        {
+            option::Option& opt = buffer[i];
+            switch (opt.index())
+            {
+                case optionIndex::HELP:
+                    // not possible, because handled further above and exits the program
+                    break;
+
+                case optionIndex::TOPIC:
+                    topic_name = std::string(opt.arg);
+                    break;
+
+                case optionIndex::DOMAIN_ID:
+                    domain = strtol(opt.arg, nullptr, 10);
+                    break;
+
+                case optionIndex::SAMPLES:
+                    count = strtol(opt.arg, nullptr, 10);
+                    break;
+
+                case optionIndex::INTERVAL:
+                    if (type == PUBLISHER)
+                    {
+                        sleep = strtol(opt.arg, nullptr, 10);
+                    }
+                    else
+                    {
+                        print_warning("publisher", opt.name);
+                    }
+                    break;
+
+                case optionIndex::WAIT:
+                    if (type == PUBLISHER)
+                    {
+                        num_wait_matched = strtol(opt.arg, nullptr, 10);
+                    }
+                    else
+                    {
+                        print_warning("publisher", opt.name);
+                    }
+                    break;
+
+                case optionIndex::ASYNC:
+                    if (type == PUBLISHER)
+                    {
+                        async = true;
+                    }
+                    else
+                    {
+                        print_warning("publisher", opt.name);
+                    }
+                    break;
+
+                case optionIndex::TRANSPORT:
+                    transport = std::string(opt.arg);
+                    break;
+
+                case optionIndex::RELIABLE:
+                    reliable = true;
+                    break;
+
+                case optionIndex::TRANSIENT_LOCAL:
+                    transient = true;
```

Comment:
add a warning in case it is not reliable

Reply:
:+1: 
Also added a warning for --samples in subscribers when durability is transient local, since the behaviour is unexpected.